composeApp/src/main/kotlin/io/github/couchtracker/ui/components/CrewCompactListItem.kt

```diff
@@ -77,42 +76,31 @@ data class CrewCompactListItemModel(
     val name: String,
     val posterModel: ImageModel?,
     val departments: Set<TmdbDepartment>,
+    val departmentsString: String?,
 ) {
     companion object {
-        suspend fun fromTmdbCrew(
-            crew: List<TmdbCrew>,
+        suspend fun <T : TmdbAnyPerson> fromTmdbCrew(
+            context: Context,
+            crew: List<T>,
             language: TmdbLanguage,
             imagePreloadOptions: ImagePreloadOptions,
+            department: (T) -> TmdbDepartment?,
         ): CrewCompactListItemModel {
             require(crew.isNotEmpty()) { "At least one crew has to be specified." }
             val base = crew.first()
             require(crew.all { it.id == base.id })
             require(crew.all { it.name == base.name })
             require(crew.all { it.profilePath == base.profilePath })
+            val departments = crew.mapNotNull { department(it) }.toSet()
             return CrewCompactListItemModel(
                 person = TmdbPerson(TmdbPersonId(base.id), language),
                 name = base.name,
-                departments = crew.mapNotNull { it.department }.toSet(),
-                posterModel = base.profilePath?.let { path ->
-                    TmdbImage(path, TmdbImageType.PROFILE).toImageModel(imagePreloadOptions)
+                departments = departments,
+                departmentsString = if (departments.isNotEmpty()) {
```

Comment:
Having to format all strings in the model seems very annoying and very boilerplate-y. Can we instead do it in the component, and simply show nothing while it's loading? We could have a component for that, e.g. `AsyncText`/`FormattedText` or something similar.